packages/presentation/presentation_containers/interfaces/unsaved_changes/initialize_unsaved_changes.ts

```diff
@@ -74,17 +74,18 @@ export const initializeUnsavedChanges = <RuntimeState extends {} = {}>(
   subscriptions.push(
     combineLatest(comparatorSubjects)
       .pipe(
-        skip(1),
         debounceTime(COMPARATOR_SUBJECTS_DEBOUNCE),
-        map((latestStates) =>
-          comparatorKeys.reduce((acc, key, index) => {
-            acc[key] = latestStates[index] as RuntimeState[typeof key];
+        combineLatestWith(lastSavedState$),
+        skip(1), // unsaved changes was initialized above - ignore the first emit
```

Comment:
I don't think we want a skip after the debounce because then other changes may be skipped if they occur during the debounce time. How about just removing the `skip` for now to fix the regression.

Reply:
Good point, I can move the `debounce` - I still think the `skip` is ideal because this impacts the initial loading of all dashboards (running the comparators once vs twice for **all** panels). What do you think?

Reply:
It seems risky since it effects all unsaved changes behavior. Maybe we could create an issue and address once embeddable refactor is complete and we have more time to explore performance optimizations. I think while we are doing the embeddable refactor, its best to limit changes outside of this scope to limit potential regressions.

Reply:
Sounds good to me! I'll create a follow up issue

Reply:
Created a follow up [here](https://github.com/elastic/kibana/issues/190335) and remove the skip in fabd6be24772447f03253fcc0e4045563a88d7fa